src/category_theory/limits/cones.lean

```diff
@@ -446,6 +462,27 @@ instance functoriality_full [full G] [faithful G] : full (functoriality F G) :=
 instance functoriality_faithful [faithful G] : faithful (functoriality F G) :=
 { map_injective' := λ X Y f g e, by { ext1, injection e, apply G.map_injective h_1 } }
 
+/--
+If `e : C ≌ D` is an equivalence of categories, then `functoriality F e.functor` induces an
+equivalence between cocones over `F` and cocones over `F ⋙ e.functor`.
+-/
+@[simps]
+def functoriality_equivalence (e : C ≌ D) : cocone F ≌ cocone (F ⋙ e.functor) :=
+let f : (F ⋙ e.functor) ⋙ e.inverse ≅ F :=
+  functor.associator _ _ _ ≪≫ iso_whisker_left _ (e.unit_iso).symm ≪≫ functor.right_unitor _ in
+{ functor := functoriality F e.functor,
+  inverse := (functoriality (F ⋙ e.functor) e.inverse) ⋙
+    (precompose_equivalence f.symm).functor,
+  unit_iso := nat_iso.of_components (λ c, cocones.ext (e.unit_iso.app _) (by tidy)) (by tidy),
+  counit_iso := nat_iso.of_components (λ c, cocones.ext (e.counit_iso.app _)
+  begin
```

Comment:
Is there a reason tidy doesn't do this one like it does the others?

Reply:
We have four related lemmas:

`counit_inv_app_functor : e.counit_inv.app (e.functor.obj X) = e.functor.map (e.unit.app X)`
`counit_app_functor : e.counit.app (e.functor.obj X) = e.functor.map (e.unit_inv.app X)`
`unit_app_inverse : e.unit.app (e.inverse.obj Y) = e.inverse.map (e.counit_inv.app Y)`
`unit_inv_app_inverse : e.unit_inv.app (e.inverse.obj Y) = e.inverse.map (e.counit.app Y)`

which one might hope would make good simp lemmas. My experiments with this so far just result in moving around the places where `tidy` (or just `simp`) disappointingly gets stuck, without reducing the number.

I've added a note in this proof explicitly admitting that it's lame to not be able to use `tidy` here.